jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -193,6 +216,56 @@ public static boolean isValidJavaClass(String className) {
     return true;
   }
 
+  /**
+   * Converts a list of port numbers to the corresponding ExposedPorts map format.
+   *
+   * <p>Example: [1000, 2000-2002] -> {"1000/tcp":{}, "2000/tcp":{}, "2001/tcp":{}, "2002/tcp":{}})
+   *
+   * @param ports the list of port numbers/ranges
+   * @return the map
+   * @throws NumberFormatException if any of the ports are in an invalid format or out of range
+   */
+  @VisibleForTesting
+  static ImmutableSortedMap<String, EmptyStruct> portListToPortMap(List<String> ports)
+      throws NumberFormatException {
+    SortedMap<String, EmptyStruct> result = new TreeMap<>();
+
+    for (String port : ports) {
+      // Remove all spaces
+      port = port.trim().replace(" ", "");
```

Comment:
What about gradle?

Reply:
In gradle, you would set something to a specific string, so it should be up to the user to make sure they type like `port = '2345'` rather than `port = ' 2345'`, and to error on the latter so they fix their stray space.